lib/filter/tests/test_filters_in_list.c

```diff
@@ -157,7 +157,8 @@ run_testcases(const char *top_srcdir)
 int
 main(int argc G_GNUC_UNUSED, char *argv[] G_GNUC_UNUSED)
 {
-  char *top_srcdir = getenv("top_srcdir");
+
+  char *top_srcdir = TOP_SRCDIR;
```

Comment:
I would split this commit (along with the Makefile change).
In my opinion, this is a bugfix-like patch and has nothing to do with CMake.